### PR TITLE
updating second snippet to call ReadToEndAsync

### LINKED ISSUE #14979: ReadToEndAsync Snippet is wrong
The code in snippet 2 is missing a line to read the streamReader

### PR DESCRIPTION
**Pull request by eddynaka**
[EDIT by guardrex to add the issue link]
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/performance/performance-best-practices?view=aspnetcore-2.1&branch=pr-en-us-14978)

Fixes #14979

### CODE REVIEW

**Comment by davidfowl** [comment]
Instead of that, let’s change the docs above the comment to explain how we can do async deserialization directly with the new JSON serializer

**Comment by eddynaka** [comment]
Fixes 14979

**Comment by davidfowl** [comment]
> @davidfowl should we add another section or we should remove that one that i updated?
> another question, for the first snippet, should we alter and dispose the streamreader?

OK lets do this, keep this change and add a 3rd snippet of DO This with the fully async and non buffered request body.

**Comment by eddynaka** [comment]
@davidfowl just added the third as you said before. 
I added the namespace in each line, because in the snippet in the docs, its not showing the namespaces used, so we won't know what nuget/library we are using to deserialize.

**Comment by guardrex** [comment]
:smile: *Now ... it throws the CLA!* @Rick-Anderson that system is still 🐞-y as heck. *Yikes!*

@eddynaka Please sign the CLA if you agree with its terms. ☝️

It's supposed to be signed before we even look at a community PR. However, that system has been a pain point for years here. It throws the CLA notice *after* making PR changes for readers and the team regularly, so it ends up breaking our process quite a bit.

File: aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs [hunk 1]
```diff
@@ -27,11 +27,25 @@ public class GoodStreamReaderController : Controller
         [HttpGet("/contoso")]
         public async Task<ActionResult<ContosoData>> Get()
         {
-            return await JsonSerializer.DeserializeAsync<ContosoData>(Request.Body);
+            var json = await new StreamReader(Request.Body).ReadToEndAsync();
+            
+            return await System.Text.Json.JsonSerializer.Deserialize<ContosoData>(json);
```
Comment (davidfowl) [inline aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs]: Remove the full namespace.

File: aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public ActionResult<ContosoData> Get()
         {           
             var json = new StreamReader(Request.Body).ReadToEnd();
 
-            return JsonSerializer.Deserialize<ContosoData>(json);
+            return System.Text.Json.JsonSerializer.Deserialize<ContosoData>(json);
```
Comment (davidfowl) [inline aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs]: Remove the full namespace.

File: aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs [hunk 3]
```diff
@@ -27,11 +27,25 @@ public class GoodStreamReaderController : Controller
...
     }
     #endregion
+    
+    #region snippet3
+    public class GoodStreamReaderController : Controller
+    {
+        [HttpGet("/contoso")]
+        public async Task<ActionResult<ContosoData>> Get()
+        {
+            return await System.Text.Json.JsonSerializer.DeserializeAsync<ContosoData>(Request.Body);
```
Comment (davidfowl) [inline aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs]: Same as above. Add a using.

File: aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs (hunk 1 above)
Comment (Rick-Anderson) [reply aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs]: Error	CS1061	'ContosoData' does not contain a definition for 'GetAwaiter' and no accessible extension method 'GetAwaiter' accepting a first argument of type 'ContosoData' could be found

Comment (eddynaka) [reply aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs]: Just updated. Thanks!

File: aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs (hunk 2 above)
Comment (eddynaka) [reply aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs]: Just updated

File: aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs (hunk 3 above)
Comment (eddynaka) [reply aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs]: Just updated

Comment (Rick-Anderson) [reply aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs]: `snippet3` needs to move to a different namespace so it compiles. It's not clear why you're adding `snippet3`

**Comment by Rick-Anderson** [comment]
@eddynaka  your branch was behind master so I had to merge them. Can you check my commits?

**Comment by eddynaka** [comment]
Thanks Rick

Everything looks good!

**Comment by eddynaka** [comment]
Fixes 14990

**Pull request merged**